components/PostCards.tsx

```diff
@@ -0,0 +1,32 @@
+import axios from 'axios';
+import React, { useEffect, useState } from 'react';
+import { BASE_API_ENDPOINT } from '../constants/api';
+import { GetAPI } from '../interfaces/api';
+import { Post } from '../interfaces/post';
+import { PostCard } from './PostCard';
+
+export function PostCards() {
+    const [posts, setPosts] = useState<Post[]>([]);
+    function sortPostsByDate(posts: Post[]): Post[] {
+        const sortedPosts = [...posts];
+        sortedPosts.sort(
+            (a, b) => new Date(b.date).getTime() - new Date(a.date).getTime(),
+        );
+        return sortedPosts;
```

Comment:
the api is already sorted, you can check the API endpoint specification.

To sort by date in descending order just add the query `ORDER=DESC`
the data is sorted in ascending order in default